scripts/review/diff.ts

```diff
@@ -0,0 +1,143 @@
+import util from 'node:util';
+import fs from 'node:fs/promises';
+import parseDiff from 'parse-diff';
+
+type Options = {
+  diffPatchFileName: string;
+  verbose: boolean;
+};
+
+enum HighestChangeType {
+  Initial,
+  Removal,
+  Addition,
+}
+
+function parseOptions(): Options {
+  const options = {
+    diffPatchFileName: { type: 'string' },
+    verbose: { type: 'boolean' },
+  } as const;
+  const { values } = util.parseArgs({ options });
+
+  const option = 'diffPatchFileName';
+  if (values[option] === undefined) {
+    throw new Error(`missing --${option} flag`);
+  }
+
+  return {
+    ...(values as Options),
+    verbose: values.verbose === true,
+  };
+}
+
+async function main() {
+  const options = parseOptions();
+  const debug = (...msg: unknown[]) => {
+    if (options.verbose) {
+      console.debug(...msg);
+    }
+  };
+
+  // Read the content of the files
+  const diffPatchContent = await fs.readFile(options.diffPatchFileName, 'utf-8');
+  const diffPatch = parseDiff(diffPatchContent);
+
+  let highestChainID = false;
+  let additionalDeploymentToSameChainID = false;
+
+  for (const file of diffPatch) {
+    const { to, from, additions, deletions } = file;
+
+    // Check to see if the file name was changed.
+    if (to !== from) {
+      throw new Error(`ERROR: ${to} and ${from} are different`);
+    }
+
+    debug(`Checking ${to} with ${additions} additions and ${deletions} deletions`);
+
+    // Check to see if the changes are valid and set any edge case flags.
+    if (additions === 2 && deletions === 1) {
+      highestChainID = true;
+    } else if (additions === 1 && deletions === 1) {
+      additionalDeploymentToSameChainID = true;
+    } else if (additions !== 1 || deletions !== 0) {
+      throw new Error(`ERROR: ${to} has invalid changes`);
+    }
+  }
+
+  if (highestChainID) {
+    debug('Highest chain ID deployment');
+    for (const file of diffPatch) {
+      const { chunks } = file;
+
+      for (const chunk of chunks) {
+        const { changes } = chunk;
+        let state = HighestChangeType.Initial;
+        let previousLine = '';
+
+        for (const change of changes) {
+          const { type, content } = change;
+
+          if (type === 'del') {
+            if (state !== HighestChangeType.Initial) {
+              throw new Error('Removal is not in correct order');
+            }
+            state = HighestChangeType.Removal;
+            previousLine = '+'.concat(content.slice(1)).concat(',');
```

Comment:
naming-nit: This is the expected next line and not the previous line right?
```suggestion
            expectedNextLine = '+'.concat(content.slice(1)).concat(',');
```

Reply:
Also, why is the `+` needed? shouldn't the diff parser already strip out the +/- prefix?

Reply:
The diff parser doesn't strip those out in `content`.